packages/big-design/src/components/Select/types.ts

```diff
@@ -0,0 +1,31 @@
+import { Placement } from 'popper.js';
+
+import { ListItemProps } from '../List/Item';
+
+export interface SelectProps extends Omit<React.HTMLAttributes<HTMLUListElement>, 'children' | 'onChange'> {
+  action?: Action;
+  disabled?: boolean;
+  error?: string;
+  label?: string;
+  maxHeight?: number;
+  multi?: boolean;
+  options: Option[];
+  placement?: Placement;
+  positionFixed?: boolean;
+  required?: boolean;
+  value?: string | number | Array<string | number>;
+  onChange(value: string | number | Array<string | number>): void;
+}
+
+interface BaseItem extends Omit<ListItemProps, 'children' | 'content' | 'onClick' | 'value'> {
+  content: string;
+  icon?: React.ReactElement;
+}
+
+export interface Option extends Omit<BaseItem, 'actionType'> {
+  value: string | number;
```

Comment:
With the new approach `value` should be able to accept any value right? Is there a reason you are limiting it to `string | number`?

Reply:
@chanceaclark and I talked about this yesterday. Since `Select` is a form element we felt it should adhere to the native behavior.

Reply:
I don't think it would complicate multi-selects, maybe we can chat about it offline?

I think it's valuable to be able to pass whatever I want as a value 🤔. Especially since you aren't doing any runtime check to verify its a string | number.

Reply:
Hmm, I haven't crossed into a case where I wanted to pass something that is not a string or a number. I would imagine that would cover the majority of use cases? It would be nice to be able to pass whatever, but is that needed for right now?